### PR TITLE
Sprint 40

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 40

### CODE REVIEW

**Comment by kfoley-18F** [comment]
We use this list when performing a code review to ensure that all tasks have been completed.

- [ ] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/204/head:sprint-43`), and then switch to that branch (e.g. `git checkout sprint-43`)
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally, with `./docker-run.sh`
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: shared/src/business/useCases/addCoversheetInteractor.test.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ describe('addCoversheetInteractor', () => {
 
   const testPdfDocBytes = () => {
     // sample.pdf is a 1 page document
-    return fs.readFileSync(testAssetsPath + 'sample.pdf');
+    return new Uint8Array(fs.readFileSync(testAssetsPath + 'sample.pdf'));
```
[the remark this answers is not in the document]
Comment (kfoley-18F) [reply shared/src/business/useCases/addCoversheetInteractor.test.js:20]: Also, I'm seeing some DRY issues with this testPdfDocBytes getting copied around several test files. 
Is it possible to make it accessible to the test harness and make it only once?

File: shared/src/business/utilities/pdfGenerator/documentTemplates/ReceiptOfFiling.test.js [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+      {
+        documentTitle: 'Secondary Supporting Document One Title',
+      },
+      {
+        documentTitle: 'Secondary Supporting Document Two Title',
+      },
+    ];
+
+    filedAt = '02/22/20 2:22am ET';
+    filedBy = 'Mike Wazowski'; // You did it, buddy. You filed your paperwork.
```
Comment (kfoley-18F) [inline shared/src/business/utilities/pdfGenerator/documentTemplates/ReceiptOfFiling.test.js:48]: 🤣 
I so ❤️  y'all making PR review fun/funny today.

File: env-manager/deleteStacks.js [hunk 3]
```diff
@@ -9,21 +9,21 @@ exports.deleteStacks = async ({ environment }) => {
     environment,
   });
   for (const stack of stacks) {
-    console.log('Delete ', stack.StackName);
+    console.log('Delete CloudFormation Stack:', stack.StackName);
     await cloudFormation.deleteStack({ StackName: stack.StackName }).promise();
-    await sleep(5000);
+    await sleep(100);
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply env-manager/deleteStacks.js:14]: You got it, I'll add units.

File: shared/src/business/useCases/addCoversheetInteractor.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthopson) [reply shared/src/business/useCases/addCoversheetInteractor.test.js:20]: @kfoley-18F `pdf-lib` requires an array buffer (and saves things as such). You'll find that we use this format with PDFs pretty much exclusively throughout the app with regard to PDF manipulation.

I've added a backlog item to DRY the usage of `testPdfDocBytes` up.

**Comment by kfoley-18F** [comment]
@JessicaMarine - I recommend that you accept this PR into staging. Thanks!

**Pull request merged**